lib/teleterm/clusters/cluster_gateways.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusters
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/lib/teleterm/gateway"
+
+	"github.com/gravitational/trace"
+)
+
+type CreateGatewayParams struct {
+	// TargetURI is the cluster resource URI
+	TargetURI string
+	// TargetUser is the target user name
+	TargetUser string
+	// LocalPort is the gateway local port
+	LocalPort string
+}
+
+// CreateGateway creates a gateway
+func (c *Cluster) CreateGateway(ctx context.Context, params CreateGatewayParams) (*gateway.Gateway, error) {
```

Comment:
Nit: I find it a little strange that we call it generic "Gateway" but it seems to be specific to databases? Maybe DatabaseGateway would be a more suitable name? For methods and structs both. I'm fine either way but wondering what will happen if we ever need to add another type of gateway.

Reply:
"Gateway" is Teleterm's name for a connection to a resource established through the ALPN proxy. For now it may appear to be specific to databases because we use it only for databases at the moment. In the future we might add support for other protocols, but they'll still be handled through gateways.

As far as I remember Alexey explaining this, the name "gateway" was chosen because there's already a lot of files dealing with "proxies", where "proxy" means different things in different contexts. So in order to avoid adding yet another place dealing with "proxies" and choose a name that will signify this specific concept (a connection made through ALPN proxy), we called them gateways in Teleterm.